src/qibocal/utils.py

```diff
@@ -0,0 +1,13 @@
+from typing import Dict, List, Union
+
+from qibolab.platform import Platform
+from qibolab.qubits import Qubit, QubitId
+
+
+def allocate_qubits(
+    platform: Platform, qubit_ids: List[QubitId]
+) -> Union[List[QubitId], Dict[QubitId, Qubit]]:
+    """Convert List[QubitId] -> Dict[QubitId, Qubit] for non-trivial platform."""
+    if platform is not None:
+        return {q: platform.qubits[q] for q in qubit_ids if q in platform.qubits}
+    return qubit_ids
```

Comment:
```suggestion
def allocate_qubits(platform: Platform) -> Dict[QubitId, Qubit]:
    """Convert List[QubitId] -> Dict[QubitId, Qubit] for non-trivial platform."""
    return {q: platform.qubits[q] for q in qubit_ids if q in platform.qubits}
```
Make it simpler: the `None` check you can do before calling the function, and in this way the function have one less argument and a well-defined return type.

Moreover, with the current implementation the `platform` type would be even *slightly wrong*: it should be `Optional[Platform]`.